palettes.cpp

```diff
@@ -247,6 +247,14 @@ std::optional<Palettes::Definition> PaletteRepository::get_palette(const std::ws
 std::vector<Palettes::Definition> PaletteRepository::get_all_custom_palettes() {
 	auto items = m_map.items();
 
+	// Remove any undefined palette indices
+	for (auto item = items.begin(); item != items.end(); item++) {
+		if (!get_palette(item->first).has_value()) {
+			remove_palette(item->first);
+			item = items.erase(item) - 1;

```

Comment:
I wanted to have this in `get_palette` but I couldn't figure out a simple way to make `std::transform()` ignore an entry.
But as this function always seems to run when doing anything with custom palettes, it should work just as well.